crates/chia-client/src/peer.rs

```diff
@@ -1,363 +1,341 @@
-
-        // Remove the one shot channel.
-        self.requests.lock().await.remove(&message_id);
-
-        // Handle the response, if present.
-        response.or(Err(Error::MissingResponse))
+        self.0.sink.lock().await.send(message).await?;
+        Ok(receiver.await?)
     }
+}
 
-    pub fn receiver(&self) -> &broadcast::Receiver<PeerEvent> {
-        &self.event_receiver
+impl Drop for PeerInner {
+    fn drop(&mut self) {
+        self.inbound_handle.abort();
     }
+}
 
-    pub fn receiver_mut(&mut self) -> &mut broadcast::Receiver<PeerEvent> {
-        &mut self.event_receiver
-    }
+async fn handle_inbound_messages(
+    mut stream: Stream,
+    sender: mpsc::Sender<Message>,
+    requests: Arc<RequestMap>,
+) -> Result<()> {
+    use tungstenite::Message::{Binary, Close, Frame, Ping, Pong, Text};
 
-    async fn handle_inbound(
-        message: WsMessage,
-        requests: &Requests,
-        event_sender: &broadcast::Sender<PeerEvent>,
-    ) -> Result<(), Error<()>> {
-        // Parse the message.
-        let message = Message::from_bytes(message.into_data().as_ref())?;
-
-        if let Some(id) = message.id {
-            // Send response through oneshot channel if present.
-            if let Some(request) = requests.lock().await.remove(&id) {
-                request.send(message).ok();
-            }
-            return Ok(());
-        }
+    while let Some(message) = stream.next().await {
+        let message = message?;
 
-        macro_rules! events {
-            ( $( $event:ident ),+ $(,)? ) => {
-                match message.msg_type {
-                    $( ProtocolMessageTypes::$event => {
-                        event_sender
-                            .send(PeerEvent::$event($event::from_bytes(message.data.as_ref())?))
-                            .ok();
-                    } )+
-                    _ => {}
-                }
-            };
+        match message {
+            Text(text) => {
+                log::warn!("Received unexpected text message: {text}");
+            }
+            Close(close) => {
+                log::warn!("Received close: {close:?}");
+                break;
+            }
+            Ping(_ping) => {}
+            Pong(_pong) => {}
+            Binary(binary) => {
+                let message = Message::from_bytes(&binary)?;
+
+                let Some(id) = message.id else {
+                    sender.send(message).await.map_err(|error| {
+                        log::warn!("Failed to send peer message event: {error}");
+                        Error::EventNotSent
+                    })?;
+                    continue;
+                };
+
+                let Some(request) = requests.remove(id).await else {
+                    log::warn!(
+                        "Received {:?} message with untracked id {id}",
+                        message.msg_type
+                    );
+                    return Err(Error::UnexpectedMessage(message.msg_type));
```

Comment:
This is disconnecting this one peer, right?

Reply:
Yeah currently this crate only handles single peer connections